docs/source/_static/rest-api.yml

```diff
@@ -859,9 +863,324 @@ paths:
+                allOf:
+                  - $ref: "#/components/schemas/PaginatedList"
+                  - type: object
+                    properties:
+                      items:
+                        type: array
+                        items:
+                          $ref: "#/components/schemas/ShareCode"
+      security:
+        - oauth2:
+            - read:shares
+    post:
+      summary: Issue share code
+      description: |
+        Issue a share code, which can be exchanged for shared access to a single server
+        (new in 5.0)
+      parameters:
+        - $ref: "#/components/parameters/sharedServerOwner"
+        - $ref: "#/components/parameters/sharedServerName"
+      requestBody:
+        description: The new share code properties, as a JSON dict.
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                expires_in:
+                  type: number
+                  description: |
+                    expiration in seconds.
+                    If unspecified, expires in one day (86400)
+                scopes:
+                  type: array
+                  description: scopes to grant
+                  items:
+                    type: string
+        required: true
+      x-codegen-request-body-name: body
+      responses:
+        200:
+          description: |
+            The Share code you just created.
+            The code itself will be in the `code` field.
+            The code is not stored by JupyterHub and cannot be retrieved a second time.
+            `accept_url` will be the actual URL to share
+            (it will look like `/hub/accept-share?code=abc123`).
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/ShareCode"
+                  - type: object
+                    properties:
+                      code:
+                        type: string
+                        description: The share code itself
+                        example: abc123
+                      accept_url:
+                        type: string
+                        description: The URL for acepting the code
+                        example: /hub/accept-share?code=abc123
+      security:
+        - oauth2:
+            - shares
+    delete:
+      summary: Revoke share code
+      description: |
+        Revoke a share code by id or code.
+        Exactly one of `id` or `code` must be specified.
+        (new in 5.0)
+      parameters:
+        - $ref: "#/components/parameters/sharedServerOwner"
+        - $ref: "#/components/parameters/sharedServerName"
+        - in: query
+          name: code
+          description: the share code to revoke
+          schema:
+            type: string
+        - in: query
+          name: id
+          description: the id of the share code to revoke
+          schema:
+            type: string
```

Comment:
Since this file is solely for the purposes of generating docs for humans, I don't think there's anything to change here. The renderer we currently use (swaggerUI) doesn't handle this multi-level schema appropriately, nor does it communicate the mutual exclusivity when configured this way.

This PR currently (clearly documents code and id as separate parameters, and that exactly one should be used):

<img width="781" alt="Screenshot 2024-01-29 at 11 10 40" src="https://github.com/jupyterhub/jupyterhub/assets/151929/cb7fc7f6-6202-4a7e-8bfd-0d242dab80f7">

With the suggested change (less clear, discards exclusivity information):

<img width="762" alt="Screenshot 2024-01-29 at 11 09 57" src="https://github.com/jupyterhub/jupyterhub/assets/151929/93c447a8-a68d-4687-b889-41afe7ae807a">

Reply:
Agreed, it's more confusing than anything else.

Thanks for the images.